src/SonarScanner.MSBuild.Shim/SonarScanner.Wrapper.cs

```diff
@@ -89,28 +90,52 @@ private static bool InternalExecute(AnalysisConfig config, IEnumerable<string> u
                 return false;
             }
 
-            var exeFileName = FindScannerExe();
+            var exeFileName = FindScannerExe(logger, Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location));
             return ExecuteJavaRunner(config, userCmdLineArguments, logger, exeFileName, fullPropertiesFilePath, new ProcessRunner(logger));
         }
 
-        private static string FindScannerExe()
+        internal /* for testing */ static string FindScannerExe(ILogger logger, string scannerCliDirectoryLocation, string scannerVersion = SonarScannerVersion)
         {
-            var binFolder = Path.GetDirectoryName(typeof(SonarScannerWrapper).Assembly.Location);
-            var fileExtension = PlatformHelper.IsWindows() ? ".bat" : "";
-            return Path.Combine(binFolder, $"sonar-scanner-{SonarScannerVersion}", "bin", $"sonar-scanner{fileExtension}");
+            var scannerCliFolder = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-{scannerVersion}");
+
+            if (!Directory.Exists(scannerCliFolder))
+            {
+                // We unzip the scanner-cli-{version}.zip while in the user's machine so that the Unix file permissions are not lost.
+                // The unzipping happens only once, during the first scanner usage.
+                var zipPath = Path.Combine(scannerCliDirectoryLocation, $"sonar-scanner-cli-{scannerVersion}.zip");
+                logger.LogInfo($"Unzipping sonar-scanner-cli-{scannerVersion}.zip");
+                if (!File.Exists(zipPath))
+                {
+                    logger.LogError($"Could not find {zipPath}");
+                }
+                // System.IO.Compression.ZipFile has zipbomb attack protection: https://github.com/dotnet/runtime/issues/15940
+                ZipFile.ExtractToDirectory(zipPath, scannerCliDirectoryLocation);
```

Comment:
I agree with @csaba-sagi-sonarsource 's point here https://github.com/SonarSource/sonar-scanner-msbuild/pull/1250/files#r895532981

it should be an if/else there

by using LogError the process should fail , and it's better to fail nicely in this case

It doesn't make sense to try to `ExtractToDirectory` if you know it will fail